src/cache.js

```diff
@@ -26,38 +26,61 @@ class Cache {
     this.models[modelName] = modelSchema(tableName)
   }
 
-  updateUserWithLoan (userID, loanID) {
-    return this.updateUserWithItem(userID, loanID, 'loan')
-  }
-
-  updateUserWithRequest (userID, requestID) {
-    return this.updateUserWithItem(userID, requestID, 'request')
-  }
-
-  updateUserWithItem (userID, itemID, itemType) {
+  updateUserItem (userID, itemID, operation, itemType) {
     return this.models.UserModel.get(userID)
       .then(user => {
         if (user) {
-          switch (itemType) {
-            case 'loan':
-              return user.addLoan(itemID).save()
-            case 'request':
-              return user.addRequest(itemID).save()
-          }
+          return {
+            'add': {
+              'loan': user.addLoan,
+              'request': user.addRequest
+            },
+            'delete': {
+              'loan': user.deleteLoan,
+              'request': user.deleteRequest
+            }
+          }[operation][itemType].call(user, itemID).save()
```

Comment:
How does this handle operation or itemType being `null` or `undefined`

Reply:
It's not really a method that's meant to be called from outside the class, it's just Javascript doesn't have proper private class methods, so it's only meant to be called internally with valid arguments even though it is public. 

Could add some sort of handling, either checking the arguments are valid or just wrap it in a try/catch. It's also possible to make it "private" in a slightly hacky way using `call`, not sure which would be better though.

Reply:
I'd be happy with adding either a try/catch or some argument validation as its been called this way.